packages/react-components/react-slider/library/docs/Spec.md

```diff
@@ -2,98 +2,311 @@
+
+#### Types
+
+```ts
+type RangeSliderOnChangeData = {
+  value: { start: number; end: number };
+};
+```
+
+### RangeSlider Structure
+
+- **Public**
+
+  ```tsx
+  <RangeSlider defaultValue={{ start: 20, end: 80 }} />
+  ```
+
+- **Internal**
+
+  ```tsx
+  <slots.root {...slotProps.root}>
+    <slots.startInput {...slotProps.startInput} />
+    <slots.endInput {...slotProps.endInput} />
+    <slots.rail {...slotProps.rail} />
+    <slots.startThumb {...slotProps.startThumb} />
+    <slots.endThumb {...slotProps.endThumb} />
+  </slots.root>
+  ```
+
+- **DOM**
+
+  ```html
+  <div class="fui-RangeSlider">
+    <input type="range" aria-hidden="true" tabindex="-1" value="20" />
+    <input type="range" aria-hidden="true" tabindex="-1" value="80" />
+    <div class="fui-RangeSlider__rail"></div>
+    <div
+      class="fui-RangeSlider__startThumb"
+      role="slider"
+      tabindex="0"
+      aria-labelledby="label-123"
+      aria-valuemin="0"
+      aria-valuemax="80"
+      aria-valuenow="20"
+      aria-valuetext="20"
+    ></div>
+    <div
+      class="fui-RangeSlider__endThumb"
+      role="slider"
+      tabindex="0"
+      aria-labelledby="label-123"
+      aria-valuemin="20"
+      aria-valuemax="100"
+      aria-valuenow="80"
+      aria-valuetext="80"
+    ></div>
+  </div>
+  ```
+
+### RangeSlider Migration
+
+RangeSlider follows the same guidance outlined in [MIGRATION.md](./MIGRATION.md).
+
+### RangeSlider Behaviors
+
+#### Keyboard
+
+| Key             | Description                                                                         |
+| --------------- | ----------------------------------------------------------------------------------- |
+| `Tab`           | Moves focus between the start thumb, end thumb, and surrounding focusable elements. |
+| `Up` / `Right`  | Increment the focused thumb by `step`.                                              |
+| `Down` / `Left` | Decrement the focused thumb by `step`.                                              |
+| `PageUp`        | Increment the focused thumb by `10 × step`.                                         |
+| `PageDown`      | Decrement the focused thumb by `10 × step`.                                         |
+| `Home`          | Move the focused thumb to its minimum allowed value.                                |
+| `End`           | Move the focused thumb to its maximum allowed value.                                |
+
+Additional logic:
+
+- Collision detection prevents thumbs from crossing. When they meet, they stop at the collision point.
+- Keyboard focus remains on the active thumb while adjusting values.
+
+#### Cursor & Touch
+
+- `pointerdown` on the rail selects the closest thumb and moves it to the pointer location.
+- `pointermove` (or touch move) continues dragging the active thumb.
+- Anti-crossing ensures thumbs never swap order. Collision is handled by clamping the active thumb to the other thumb's position.
+
+#### Screen readers
```

Comment:
maybe we should also mention here that when `vertical` is true we expose also `aria-orientation`: 
- https://w3c.github.io/aria/#aria-orientation
- https://www.w3.org/WAI/ARIA/apg/patterns/slider-multithumb/